cmd/gateway-gcs.go

```diff
@@ -631,6 +633,9 @@ func (l *gcsGateway) ListMultipartUploads(bucket string, prefix string, keyMarke
 }
 
 func fromGCSMultipartKey(s string) (key, uploadID string, partID int, err error) {
+	// remove prefixes
+	s = filepath.Base(s)
```

Comment:
Yes it won't we need to always use path.Base() not filepath.  All incoming object names are expected to be separated by "/" not by "\\" even on windows.